promlog/log.go

```diff
@@ -106,3 +124,52 @@ func New(config *Config) log.Logger {
 	l = log.With(l, "ts", timestampFormat, "caller", log.DefaultCaller)
 	return l
 }
+
+// NewDynamic returns a new leveled logger. Each logged line will be annotated
+// with a timestamp. The output always goes to stderr. Some properties can be
+// changed, like the level.
+func NewDynamic(config *Config) *logger {
+	var l log.Logger
+	if config.Format != nil && config.Format.s == "json" {
+		l = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
+	} else {
+		l = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	}
+	l = log.With(l, "ts", timestampFormat, "caller", log.DefaultCaller)
+
+	lo := &logger{
+		base:    l,
+		leveled: l,
+	}
+	if config.Level != nil {
+		lo.SetLevel(config.Level)
+	}
+	return lo
+}
+
+type logger struct {
+	base         log.Logger
+	leveled      log.Logger
+	currentLevel *AllowedLevel
+	mtx          sync.Mutex
+}
+
+// l implements logger.Log
```

Comment:
Is this the correct syntax for this godoc?

Reply:
It is not, but I had fixed it. Where did I push...